File: Ahorn/entities/breakablePot.jl
module JungleHelperBreakablePot

using ..Ahorn, Maple

@mapdef Entity "JungleHelper/BreakablePot" BreakablePot(x::Integer, y::Integer, sprite::String="", rupeeImage::String="JungleHelper/Breakable Pot/rupee")

const placements = Ahorn.PlacementDict(
    "Breakable Pot (Jungle Helper)" => Ahorn.EntityPlacement(
        BreakablePot
    )
)

function Ahorn.selection(entity::BreakablePot)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 8, y - 20, 17, 20)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BreakablePot, room::Maple.Room) = Ahorn.drawSprite(ctx, "JungleHelper/Breakable Pot/breakpotidle", 0, -9)

end

File: Loenn/entities/breakablePot.lua
local utils = require("utils")

local breakablePot = {}
breakablePot.name = "JungleHelper/BreakablePot"
breakablePot.depth = 1
breakablePot.placements = {
    {
        name = "default",
        data = {
            sprite = "",
            rupeeImage = "JungleHelper/Breakable Pot/rupee",
            containsKey = false,
        }
    }
}

breakablePot.offset = { 12, 21 }
breakablePot.texture = "JungleHelper/Breakable Pot/breakpotidle"

function breakablePot.selection(room, entity)
    return utils.rectangle(entity.x - 8, entity.y - 20, 17, 20)
end

return breakablePot
